app/src/main/java/com/example/stromtracker/database/Geraete.kt

```diff
@@ -3,112 +3,134 @@ package com.example.stromtracker.database
 
 import androidx.room.*;
 
-@Entity
-data class Geraete(
-    @ColumnInfo(name = "name") private var name: String,
-    @ColumnInfo(name = "kategorieID") private var kategorieID: Int,
-    @ColumnInfo(name = "raumID") private var raumID: Int,
-    @ColumnInfo(name = "haushaltID") private var haushaltID: Int,
-    @ColumnInfo(name = "stromVollast") private var stromVollast: Int,
-
-    @ColumnInfo(name = "stromStandBy") private var stromStandBy: Int,
-    @ColumnInfo(name = "betriebszeit") private var betriebszeit: Int,
-    @ColumnInfo(name = "urlaubsmodus") private var urlaubsmodus: Boolean,
-    @ColumnInfo(name = "notiz") private var notiz: String?
-
-) {
-    @PrimaryKey(autoGenerate = true) private var geraeteID: Int = 0
-
-    fun getGeraeteID():Int {
-        return geraeteID
-    }
+@Entity (indices = [ Index("kategorieID"), Index("haushaltID"), Index("raumID")  ], foreignKeys = [
+    ForeignKey(
+        entity = Kategorie::class,
+        parentColumns = arrayOf("kategorieID"),
+        childColumns = arrayOf("kategorieID"),
+        onDelete = ForeignKey.NO_ACTION),
+    ForeignKey(
+        entity = Haushalt::class,
+    parentColumns = arrayOf("haushaltID"),
+    childColumns = arrayOf("haushaltID"),
+        onDelete = ForeignKey.CASCADE
+    ),
+    ForeignKey(
+        entity = Raum::class,
+        parentColumns = arrayOf("raumID"),
+        childColumns = arrayOf("raumID"),
+        onDelete = ForeignKey.NO_ACTION
+)
+])
+    data class Geraete(
+        @ColumnInfo(name = "name") private var name: String,
+        @ColumnInfo(name = "kategorieID") private var kategorieID: Int,
+        @ColumnInfo(name = "raumID") private var raumID: Int,
+        @ColumnInfo(name = "haushaltID") private var haushaltID: Int,
+        @ColumnInfo(name = "stromVollast") private var stromVollast: Int,
 
-    fun setGeraeteID(v:Int) {
-        geraeteID = v
-    }
+        @ColumnInfo(name = "stromStandBy") private var stromStandBy: Int,
+        @ColumnInfo(name = "betriebszeit") private var betriebszeit: Int,
+        @ColumnInfo(name = "urlaubsmodus") private var urlaubsmodus: Boolean,
+        @ColumnInfo(name = "notiz") private var notiz: String?
+)
 
-    fun getName():String {
-        return name
-    }
 
-    fun setName(v:String) {
-        name = v
-    }
 
-    fun getStromVollast():Int {
-        return stromVollast
-    }
+    {
+        @PrimaryKey(autoGenerate = true) private var geraeteID: Int = 0
```

Comment:
Du hast jetzt gesagt, dass die Id´s ja automatisch generiert werden und setzt jetzt hier int auf 0. Ich dachte, die Primary keys beginnen bei 1 oder wofür genau ist das 0 da?

Reply:
Man braucht bei Kotlin irgendeinen Wert zum initialisieren, man kann nichts auf "null" setzen.
Diese 0 bedeutet im Endeffekt also nichts, die starten trotzdem ganz normal bei 1 und zählen dann hoch.

Reply:
Also zumindest wenn man es auf not null setzt, was ja bei einem Primärschlüssel der Fall ist.